src/hypertable.c

```diff
@@ -3014,3 +3020,112 @@ ts_hypertable_update_dimension_partitions(const Hypertable *ht)
 
 	return false;
 }
+
+/*
+ * hypertable_osm_range_update
+ * 0 hypertable REGCLASS,
+ * 1 range_start,
+ * 2 range_end,
+ * 3 empty=false
+ * If empty is set to true then the range will be set to invalid range
+ * but the overlap flag will be unset, indicating that no data is managed
+ * by OSM and therefore timescaledb optimizations can be applied.
+ */
+TS_FUNCTION_INFO_V1(ts_hypertable_osm_range_update);
+Datum
+ts_hypertable_osm_range_update(PG_FUNCTION_ARGS)
+{
+	Oid relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Hypertable *ht;
+	const Dimension *time_dim;
+	Cache *hcache;
+
+	Oid time_type; /* required for resolving the argument types, should match the hypertable
+					  partitioning column type */
+
+	hcache = ts_hypertable_cache_pin();
+	ht = ts_resolve_hypertable_from_table_or_cagg(hcache, relid, true);
+	Assert(ht != NULL);
+	time_dim = hyperspace_get_open_dimension(ht->space, 0);
+
+	Ensure(time_dim != NULL,
```

Comment:
Ensure means this will coredump in debug builds which i dont think is the correct behaviour

Reply:
In many other places `time_dim != NULL` is an assertion, hence the Ensure. Not finding a time dimension seems very extreme to me so I think it made sense for this to be an Ensure, but if you think throwing a simple error here is better, I'll update this.

Reply:
The internal compression hypertable has no dimensions, so it would be easy to trigger that way. If we catch this case in the lookup before i'm fine with that. Maybe also add a test calling it with internal hypertable to confirm error handling works correctly.

Reply:
Updated this to throw an error instead and added test